controllers/s3utils.go

```diff
@@ -252,6 +255,82 @@ func (s *s3ObjectStore) createBucket(bucket string) (err error) {
 	return nil
 }
 
+// deleteBucket deletes the S3 bucket.  Fails to delete if the bucket contains
+// any objects.
+func (s *s3ObjectStore) deleteBucket(bucket string) (
+	err error) {
+	if bucket == "" {
+		return fmt.Errorf("empty bucket name for "+
+			"endpoint %s caller %s", s.s3Endpoint, s.callerTag)
+	}
+
+	defer func() {
+		if r := recover(); r != nil {
+			// change the named return err value
+			err = fmt.Errorf("delete bucket recovered for %s, with %v",
+				bucket, r)
+		}
+	}()
+
+	dbInput := &s3.DeleteBucketInput{Bucket: &bucket}
+	if err = dbInput.Validate(); err != nil {
+		return fmt.Errorf("delete bucket input validation failed for %s, err %w",
+			bucket, err)
+	}
+
+	_, err = s.client.DeleteBucket(dbInput)
+	if err != nil && !isAwsErrCodeNoSuchBucket(err) {
+		return fmt.Errorf("failed to delete bucket %s, %w",
+			bucket, err)
+	}
+
+	return nil
+}
+
+// purgeBucket empties the content of the given bucket.
```

Comment:
(clarification) This is a non-recursive purge of the bucket, right? IOW if it had sub buckets it would fail (is that even a thing?).

Also, comment can state "purges and deletes the bucket..."

Reply:
You can't create a bucket from within another bucket. Sure, will update the comment.